src/cryptography/hazmat/backends/openssl/rsa.py

```diff
@@ -559,6 +565,62 @@ def private_numbers(self):
             )
         )
 
+    def as_bytes(self, encoding, format, encryption_algorithm):
+        if not isinstance(encoding, Encoding):
+            raise TypeError("encoding must be an item from the Encoding enum")
```

Comment:
We don't really do that /anywhere/ else right now. If we want to do that we
should do it everywhere IMO. (Also, this is trivial to debug with sentry,
just sayin)

On Sat, Feb 28, 2015 at 12:50 PM, Glyph <notifications@github.com> wrote:

> In src/cryptography/hazmat/backends/openssl/rsa.py
> <https://github.com/pyca/cryptography/pull/1503#discussion_r25561072>:
>
> > @@ -559,6 +565,62 @@ def private_numbers(self):
> >              )
> >          )
> >
> > +    def as_bytes(self, encoding, format, encryption_algorithm):
> > +        if not isinstance(encoding, Encoding):
> > +            raise TypeError("encoding must be an item from the Encoding enum")
>
> Experience has taught me that any error message like this *really* ought
> to have an "… instead got …" portion which explains what object was
> received instead. Otherwise debugging can be agonizing - "I only get this
> exception in production, my unit tests aren't catching it, what am I doing?
> am I reversing the order of arguments, is something naming a variable
> wrong..."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1503/files#r25561072>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084